src/core/data.jl

```diff
@@ -1,23 +1,242 @@
+                push!(repairs, (comp_name, comp_id))
             end
+        end
+    end
+    return repairs
+end
+
+
+"Count the number of items with a key \"damaged\" == 1 in a network"
+function count_damaged_items(network::Dict{String, Any})
+    if _IMs.ismultinetwork(network)
+        Memento.warn(_PMs._LOGGER, "count_damaged_items supports single networks.  Attempting to select network 0.")
+        if haskey(network["nw"],"0")
+            Memento.info(_PMs._LOGGER, "Network 0 found.")
+            nw = network["nw"]["0"]
         else
-            for i in damaged_items[id]
-                nw_data[id][i]["damaged"] = 1
+            Memento.error(_PMs._LOGGER, "Network 0 not found.")
+        end
+    else
+        nw = network
+    end
+
+    damage_count = 0
+    for (comp_name, status_key) in _PMs.pm_component_status
+        for (i, comp) in get(nw, comp_name, Dict())
+            if haskey(comp, "damaged") && comp["damaged"] == 1
+                damage_count += 1
             end
         end
     end
+    
+    return damage_count
 end
 
+
+""
+function get_damaged_items(data::Dict{String,Any})
+    if _IMs.ismultinetwork(data)
+        comp_list = Dict{String,Any}("nw" => Dict{String,Any}())
+        for (i, nw_data) in data["nw"]
+            comp_list["nw"][i] = _get_damaged_items(nw_data)
+        end
+    else
+        comp_list = _get_damaged_items(data)
+    end
+    return comp_list
+end
+
+
+""
+function _get_damaged_items(network::Dict{String,Any})
+    comp_list = Array{Tuple{String,String},1}()
+    for (comp_type, comp_status) in _PMs.pm_component_status
+        for (comp_id, comp) in network[comp_type]
+            if haskey(comp, "damaged") && comp["damaged"] == 1
+                push!(comp_list, (comp_type,comp_id))
+            end
+        end
+    end
+    return comp_list
+end
+
+
+""
+function get_isolated_load(data::Dict{String,Any})
+    if _IMs.ismultinetwork(data)
+        load_list = Dict{String,Any}("nw" => Dict{String,Any}())
+        for (i, nw_data) in data["nw"]
+            load_list["nw"][i] = _get_isolated_load(nw_data)
+        end
+    else
+        load_list = _get_isolated_load(data)
+    end
+    return load_list
+end
+
+
+""
+function _get_isolated_load(network::Dict{String,Any})
+    load_list = Array{Tuple{String,String},1}()
+
+    bus_status =  _PMs.pm_component_status["bus"]
+    bus_inactive = _PMs.pm_component_status_inactive["bus"]
+
+    load_status = _PMs.pm_component_status["load"]
+    load_inactive = _PMs.pm_component_status_inactive["load"]
+
+        for (load_id, load) in network["load"]
+            if haskey(network["bus"]["$(load["load_bus"])"], bus_status) &&  network["bus"]["$(load["load_bus"])"][bus_status] == bus_inactive
+                push!(load_list, ("load", load_id))
+            end
+        end   
+    
+    return load_list
+end
+
+
+""
+function set_component_inactive!(data::Dict{String,Any}, comp_list::Array{Tuple{String,String},1})
```

Comment:
Same note as above about the encoding of the `comp_list`.

Reply:
That's a good idea